gen/o365

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/perl
+
+# Generates members of o365 service in JSON format
+
+use strict;
+use warnings;
+use perunServicesInit;
+use perunServicesUtils;
+use JSON::XS;
+
+sub processMembers;
+
+our $SERVICE_NAME     = "o365";
+our $PROTOCOL_VERSION = "3.0.0";
+my $SCRIPT_VERSION = "3.0.0";
+
+perunServicesInit::init;
+my $DIRECTORY = perunServicesInit::getDirectory;
+my $data      = perunServicesInit::getHierarchicalData;
+
+our $A_USER_FIRST_NAME;         *A_USER_FIRST_NAME =            \'urn:perun:user:attribute-def:core:firstName';
+our $A_USER_LAST_NAME;          *A_USER_LAST_NAME =             \'urn:perun:user:attribute-def:core:lastName';
+our $A_USER_DISPLAY_NAME;       *A_USER_DISPLAY_NAME =          \'urn:perun:user:attribute-def:core:displayName';
+our $A_USER_PREFERRED_MAIL;     *A_USER_PREFERRED_MAIL =        \'urn:perun:user:attribute-def:def:preferredMail';
+our $A_USER_USER_TYPE;          *A_USER_USER_TYPE =             \'urn:perun:user:attribute-def:def:o365-userType';
+our $A_USER_ASSIGNED_LICENSES;  *A_USER_ASSIGNED_LICENSES =     \'urn:perun:user:attribute-def:def:o365-assignedLicenses';
```

Comment:
@Laliska: It's mainly for usability reasons. You create multiple resources, each with different licence, e.g. one resource with student licence and one with regular licence. Then you assign students group to students resource and employees to regular licence resource. (If student is employee at the same time, he gets both licences, which is desirable). Easy to use, easy to explain.

If the licence setting is stored in user attribute, you have to set this attribute individually for each user, which does not scale. Moreover you can delegate right to manage group membership and by that you can delegate rights to manage only one license. To be able to do this with attribute, you have to have sophisticated authorization framework over attributes values which we don't have.